### DISCUSSION TITLE
how to display some surface or show the trajectory of end effector in icub simulator

### DISCUSSION THREAD

**Discussion by YANGGANG1971** [robotology/community#436] [Q&A] [answered]
Is that possible to display some surface or show the trajectory of the end effector in icub simulator?
We currently are using iCub sim to draw some curve or trajectory, but we cannot be 100 percent sure that whether iCub sim is drawing as we expected. So do you have any ideas letting us clearly see the trajectory or curve that icub is drawing or doing?  

![image](https://user-images.githubusercontent.com/55526431/94248571-a2663180-ff59-11ea-80d5-7a3fd38e71ca.png)

**Comment by pattacini** [comment, marked as answer]
Hi @YANGGANG1971 

`iCub_SIM` does not currently provide what you need. Instead, take a look at the [`iCubGui`][1] belonging to our `icub-main` codebase, which shows a pictorial representation of the robot it connects to (simulated or physical).

Within iCubGui, you're allowed to plot simple geometric trajectories as done for example here: https://github.com/robotology/d4c#the-cartesian-if-implementation.

![](https://github.com/robotology/d4c/raw/master/img/icubgui.jpg)

Relevant code snippets can be browsed in the same repository: https://github.com/robotology/d4c/blob/master/lib/src/d4c_server.cpp#L1995-L2054.

[1]: https://robotology.github.io/robotology-documentation/doc/html/group__icub__gui.html
